test/libyul/yulOptimizerTests/fullSuite/mcopy_redundant_same_area.sol

```diff
@@ -0,0 +1,19 @@
+{
+    calldatacopy(0, 0, 0x20)
+
+    mcopy(0, 0, 0x20) // Redundant. Does not change any values (only affects MSIZE).
```

Comment:
Honestly, I haven't verified this so I could be wrong, but I assume it still is as long as your code does not use `msize` (and using that basically disables the optimizer). If you're not using `msize` then changing how much the memory is expanded only really changes gas usage (and potentially whether you contract will revert with out of gas or not). These are fair game for optimization.

Still, in this example the `mcopy` instruction does not really affect `msize` since `calldatacopy` already expands memory. The comment was more to indicate that it is the only thing `mcopy` of this form could affect. I'll remove that bit to avoid confusion.